### ISSUE TITLE
(feat): Speculative decoding

### ISSUE THREAD

**Issue by mariannearr** [kuleshov-group/dllm-dev#47]
- Keep encoder logits for NTP
